dmoj/cptbox/landlock_helper.c

```diff
@@ -0,0 +1,58 @@
+#include "landlock_helpers.h"
+
+#if PTBOX_LANDLOCK
+    #include <linux/landlock.h>
+
+    int add_rules(const int ruleset_fd, const char* const* const paths, __u64 access_rule) {
+        struct landlock_path_beneath_attr path_beneath = {
+            .parent_fd = -1,
+            .allowed_access = access_rule,
+        };
+
+        for(const char* const* pathptr = paths; *pathptr; pathptr++) {
```

Comment:
Where is this array being null-terminated? Is this relying on `alloc_byte_array` ending in a null?

Reply:
Yes